internal/metrics/disk_stats_helper.go

```diff
@@ -0,0 +1,52 @@
+package metrics
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+)
+
+type DiskStatsHelper interface {
+	GetDiskUsage(directory string) (usedBytes uint64, freeBytes uint64, err error)
+}
+
+type OSDiskStatsHelper struct {
+}
+
+//go:generate mockgen -source=disk_stats_helper.go -package=metrics -destination=mock_disk_stats_helper.go
+func NewOSDiskStatsHelper() *OSDiskStatsHelper {
+	return &OSDiskStatsHelper{}
+}
+
+func (c *OSDiskStatsHelper) getUsedBytesInDirectory(directory string) (uint64, error) {
+	var totalBytes uint64
+	err := filepath.Walk(directory, func(path string, fileInfo os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !fileInfo.IsDir() {
+			totalBytes += uint64(fileInfo.Size())
```

Comment:
@paul-maidment Would this count twice hard links?

Reply:
I think we need to account for it

https://github.com/openshift/assisted-service/pull/7217/files